### ISSUE TITLE
Drilldown with parent links doesn't allow for proper use of keyboard 'Enter' key

### ISSUE THREAD

**Issue by dbombine** [zurb/foundation-sites#10803]
#### How to reproduce this bug:

1. Create a basic drilldown component with data-parent-link set to true.
2. Add a valid href attribute to a submenu parent item.
3. Navigate to that parent item using keyboard. Press 'Enter'.
4. Navigate to the newly opened submenu parent item with a valid href. Press 'Enter' again.
 
#### What should happen:
1. When the submenu opens, the focus should be set on the submenu parent.
2. Pressing 'Enter' on the submenu parent item should take you to the location in its href attribute.

#### What happened instead:
1. When the submenu opens, focus is set incorrectly on the first submenu child.
2. Pressing 'Enter' on the submenu parent item makes it act like the 'Back' button.

#### Browser(s) and Device(s) tested on:
All browsers.

#### Foundation Version(s) you are using:
Any version. Codepen example is using 6.3.1.

#### Test case link:

- [CodePen with Foundation 6.3.1](https://codepen.io/dbombine/pen/XgwOrZ)

**Comment by Owlbertz** [comment]
Yup, that seems indeed like a bug. I will check this.
